src/components/events.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StaticQuery, graphql } from 'gatsby';
+import { contentExists } from '../utils/ug-utils.js';
+import moment from 'moment';
+import '../styles/events.css';
+
+function findMatches(fromDrupal, fromWP) {
+    for (let i=0; i < fromDrupal.length; i++) {
+        for (let j=0; j < fromWP.length; j++) {
+            return fromDrupal[i].name === fromWP[j].name;
+        }
+    }
+}
+
+const generateEvents = (data, eventData) => {
+    let drupalCategories = eventData.relationships.field_event_category;
+    let matchAllCategories = eventData.field_match_categories;
+    let title = eventData.field_title;
+    let allEvents = data.allWpEvent.edges;
+    let shownEvents = [];
+    
+    if (matchAllCategories) {
+        for (let i=0; i < allEvents.length; i++) {        
+            if (JSON.stringify(drupalCategories) === JSON.stringify(allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }        
+        } 
+    } else {
+        for (let i=0; i < allEvents.length; i++) {
+            if (findMatches(drupalCategories, allEvents[i].node.eventsCategories.nodes)) {
+                shownEvents.push(allEvents[i]);
+            }
+        } 
+    }
+    
+    return (contentExists(shownEvents) ?
+    <React.Fragment key={eventData.drupal_id}>
+        <h2 className="mb-5">{contentExists(title) ? title : "Upcoming Events"}</h2>
+        <div className="event-list row gx-3 gy-5 justify-content-md-center mb-5">
+        {shownEvents.slice(0,4).map(wpEvent => {
+            let eventMonth = moment(wpEvent.node.startDate,"YYYY-MM-DD").format("MMM");
+            let eventDay = moment(wpEvent.node.startDate,"YYYY-MM-DD").format("D");
+            let eventStartTime = moment(wpEvent.node.startDate,"YYYY-MM-DD HH:mm").format("h:mm A");
+            let eventEndTime = moment(wpEvent.node.endDate,"YYYY-MM-DD HH:mm").format("h:mm A");
+            
+            return (<>
+            <div key={wpEvent.node.id} className="card border-0 col-lg-3 col-md-6 flex-row">
+                <div className="event-day col-4 border border-5 d-flex me-3 p-2">
+                    <p className="align-self-center mb-0 mx-auto text-center w-50">
+                        <span className="fs-2 text-nowrap text-uppercase">{eventMonth}</span> <span className="display-4 fw-bold">{eventDay}</span>

```

Comment:
See https://www.accessibilityoz.com/2021/06/the-trouble-with-blind-dates-screen-readers-and-date-formatting/ - it may not be possible to control how every screen-reading app chooses to read times. I've tried a different fix for the month and day, however, so see how it reads now

Reply:
I think your solution works! :) I do hear the date twice in VoiceOver, but it is very clear to me what date and time the event occurs on.